streams/src/main/java/org/apache/kafka/streams/processor/internals/RepartitionTopicConfig.java

```diff
@@ -70,7 +76,8 @@ public boolean equals(final Object o) {
         }
         final RepartitionTopicConfig that = (RepartitionTopicConfig) o;
         return Objects.equals(name, that.name) &&
-               Objects.equals(topicConfigs, that.topicConfigs);
+               Objects.equals(topicConfigs, that.topicConfigs) &&
+               Objects.equals(numberOfPartitions, that.numberOfPartitions);
```

Comment:
If we add this to `equals`, we *must* add it to `hashCode` as well, and we _should_ also add it to `toString()`.